hrjobcontract/tests/phpunit/CRM/Hrjobcontract/BAO/HRJobDetailsTest.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use Civi\Test\TransactionalInterface;
+
+/**
+ * Class CRM_Hrjobcontract_BAO_HRJobDetailsTest
+ *
+ * @group headless
+ */
+class CRM_Hrjobcontract_BAO_HRJobDetailsTest extends PHPUnit_Framework_TestCase implements
+  HeadlessInterface,
+  TransactionalInterface {
+
+  use HRJobContractTestTrait;
+
+  public function setUpHeadless() {
+    return \Civi\Test::headless()->installMe(__DIR__)->apply();
+  }
+
+  function setUp() {
+    $upgrader = CRM_Hrjobcontract_Upgrader::instance();
+    $upgrader->install();
+  }
+
+  function testValidateDatesWithDifferentDateSets() {
+    // Create test Contact.
+    $contactParams = array("first_name" => "chrollo", "last_name" => "lucilfer");
+    $contactID =  $this->createContact($contactParams);
+
+    // Create Two test Job Contracts.
+    $this->createJobContract($contactID, '2016-01-01', '2016-01-10');
+    $this->createJobContract($contactID, '2016-02-01');
+
+    // Set of tests against invalid dates period (there are Job Contracts
+    // already with period dates overlapping.
+
```

Comment:
I think it would be better to use a [Data Provider](https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers) instead of putting the array inside the test method. It would also make thins clearer